libs/agno/agno/team/team.py

```diff
@@ -4040,94 +4000,46 @@ def determine_tools_for_model(
-                files=files,  # type: ignore
-                knowledge_filters=knowledge_filters,
-                workflow_context=workflow_context,
-                store_member_responses=store_member_responses,
-                debug_mode=debug_mode,
-                add_history_to_context=add_history_to_context,
-                dependencies=dependencies,
-            )
-            _tools.append(forward_task_func)
-            if self.get_member_information_tool:
-                _tools.append(self.get_member_information)
-
-        elif self.mode == "coordinate":
-            _tools.append(
-                self._get_delegate_task_function(
-                    run_response=run_response,
-                    session=session,
-                    session_state=session_state,
-                    team_run_context=team_run_context,
-                    user_id=user_id,
-                    stream=self.stream or False,
-                    stream_intermediate_steps=self.stream_intermediate_steps,
-                    async_mode=async_mode,
-                    images=images,  # type: ignore
-                    videos=videos,  # type: ignore
-                    audio=audio,  # type: ignore
-                    files=files,  # type: ignore
-                    knowledge_filters=knowledge_filters,
-                    workflow_context=workflow_context,
-                    store_member_responses=store_member_responses,
-                    debug_mode=debug_mode,
-                    add_history_to_context=add_history_to_context,
-                    dependencies=dependencies,
-                )
-            )
-            if self.get_member_information_tool:
-                _tools.append(self.get_member_information)
 
-        elif self.mode == "collaborate":
-            run_member_agents_func = self._get_run_member_agents_function(
-                run_response=run_response,
-                session=session,
-                session_state=session_state,
-                team_run_context=team_run_context,
-                user_id=user_id,
-                stream=self.stream or False,
-                stream_intermediate_steps=self.stream_intermediate_steps,
-                async_mode=async_mode,
-                images=images,  # type: ignore
-                videos=videos,  # type: ignore
-                audio=audio,  # type: ignore
-                files=files,  # type: ignore
-                workflow_context=workflow_context,
-                store_member_responses=store_member_responses,
-                debug_mode=debug_mode,
-                add_history_to_context=add_history_to_context,
-            )
-            _tools.append(run_member_agents_func)
+        delegate_task_func = self._get_delegate_task_function(
+            run_response=run_response,
+            session=session,
+            session_state=session_state,
+            team_run_context=team_run_context,
+            input=user_message,
```

Comment:
@dirkbrnd here still i think if a structured input is given to Team, it would be processed in the `_get_user_message` function to a `Message` object which i think model dumps json and stringifies the pydantic model if i'm not wrong. What the linux foundation engineer mentioned that they probably want to have the structured input pass as is to the member agent is still missing right?